geomesa-kafka/geomesa-kafka-datastore/src/main/scala/org/locationtech/geomesa/kafka/index/FeatureStateFactory.scala

```diff
@@ -54,7 +54,7 @@ object FeatureStateFactory extends LazyLogging {
       // remove tasks when canceled, otherwise they will only be removed from the task queue
       // when they would be executed. we expect frequent cancellations due to feature updates
       es.setRemoveOnCancelPolicy(true)
-      (es, Ticker.systemTicker())
```

Comment:
yeah, Ticker.systemTicker returns System.nanoTime, which is not actually the current time: https://docs.oracle.com/javase/7/docs/api/java/lang/System.html#nanoTime()

> Returns the current value of the running Java Virtual Machine's high-resolution time source, in nanoseconds.
This method can only be used to measure elapsed time and is not related to any other notion of system or wall-clock time. The value returned represents nanoseconds since some fixed but arbitrary origin time (perhaps in the future, so values may be negative). The same origin is used by all invocations of this method in an instance of a Java virtual machine; other virtual machine instances are likely to use a different origin.
> The values returned by this method become meaningful only when the difference between two such values, obtained within the same instance of a Java virtual machine, is computed.

I was under the impression it was the same as System.currentTimeMillis except in nanos. That's what the new CurrentTimeTicker returns.

Reply:
Wow.  Good to know!